include/boost/date_time/date_parsing.hpp

```diff
@@ -61,7 +60,7 @@ namespace date_time {
     inline unsigned short
     month_str_to_ushort(std::string const& s) {
       if((s.at(0) >= '0') && (s.at(0) <= '9')) {
-        return boost::lexical_cast<unsigned short>(s);
```

Comment:
so my suggestion is we write the function 
```
#ifdef BOOST_NO_CXX11
inline unsigned short string_to_ushort(...)
{
  --> current lexical cast
}
#else     
inline unsigned short string_to_ushort(...)
{
  --> current stoul code
}
#endif
```

This can live here or if it's used elsewhere I'd move it to compiler_config.hpp since its a work around